src/libraries/System.Text.Json/src/System/Text/Json/Serialization/Metadata/JsonTypeInfo.cs

```diff
@@ -595,6 +553,18 @@ internal void Configure()
                 }
             }
 
+            if (ElementType != null)
+            {
+                _elementTypeInfo ??= Options.GetTypeInfoInternal(ElementType);
+                _elementTypeInfo.EnsureConfigured();
+            }
+
+            if (KeyType != null)
+            {
+                _keyTypeInfo ??= Options.GetTypeInfoInternal(KeyType);
+                _keyTypeInfo.EnsureConfigured();
+            }
```

Comment:
can we add more tests with recursive types i.e.:
```csharp
class RecursiveType
{
  public List<RecursiveType> Children { get; set; }
}
```

and similar?

Reply:
We do have tests for types like that already.